src/components/core/PivotTable.tsx

```diff
@@ -123,167 +117,21 @@ export interface IPivotTableState {
-        dimensions[searchDimensionIndex],
-    ]);
-
-    if (lastFieldType === FIELD_TYPE_ATTRIBUTE) {
-        for (const header of attributeHeaders) {
-            if (getIdsFromUri(header.attributeHeader.uri)[0] === lastFieldId) {
-                return {
-                    attributeSortItem: {
-                        direction,
-                        attributeIdentifier: header.attributeHeader.localIdentifier,
-                    },
-                };
-            }
-        }
-        invariant(false, `could not find attribute header matching ${colId}`);
-    } else if (lastFieldType === FIELD_TYPE_MEASURE) {
-        const headerItem = measureHeaderItems[parseInt(lastFieldId, 10)];
-        const attributeLocators = fields.slice(0, -1).map((field: string[]) => {
-            // first item is type which should be always 'a'
-            const [, fieldId, fieldValueId] = field;
-            const attributeHeaderMatch = attributeHeaders.find(
-                (attributeHeader: Execution.IAttributeHeader) => {
-                    return getIdsFromUri(attributeHeader.attributeHeader.formOf.uri)[0] === fieldId;
-                },
-            );
-            invariant(
-                attributeHeaderMatch,
-                `Could not find matching attribute header to field ${field.join(ID_SEPARATOR)}`,
-            );
-            return {
-                attributeLocatorItem: {
-                    attributeIdentifier: attributeHeaderMatch.attributeHeader.localIdentifier,
-                    element: `${attributeHeaderMatch.attributeHeader.formOf.uri}/elements?id=${fieldValueId}`,
-                },
-            };
-        });
-        return {
-            measureSortItem: {
-                direction,
-                locators: [
-                    ...attributeLocators,
-                    {
-                        measureLocatorItem: {
-                            measureIdentifier: headerItem.measureHeaderItem.localIdentifier,
-                        },
-                    },
-                ],
-            },
-        };
-    }
-    invariant(false, `could not find header matching ${colId}`);
-};
-
-export interface ISortModelItem {
-    colId: string;
-    sort: AFM.SortDirection;
-}
-
-export const getSortsFromModel = (
-    sortModel: ISortModelItem[], // AgGrid has any, but we can do better
-    execution: Execution.IExecutionResponses,
-) => {
-    return sortModel.map((sortModelItem: ISortModelItem) => {
-        const { colId, sort } = sortModelItem;
-        const sortHeader = getSortItemByColId(execution, colId, sort);
-        invariant(sortHeader, `unable to find sort item by field ${colId}`);
-        return sortHeader;
-    });
-};
-
+/**
+ * This component is passed to AG-Grid and will be used to render loading indicator on row-level
+ *
+ * @param props
+ * @constructor
+ */
 export const RowLoadingElement = (props: ICellRendererParams) => {
```

Comment:
This could also be extracted to its own separate file.

Reply:
moved to separate component, moved tests as well